File: app/src/main/java/me/imzack/app/end/injector/module/HomePresenterModule.java
package me.imzack.app.end.injector.module;

import me.imzack.app.end.view.contract.HomeViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class HomePresenterModule {

    private final HomeViewContract mHomeViewContract;

    public HomePresenterModule(HomeViewContract homeViewContract) {
        mHomeViewContract = homeViewContract;
    }

    @Provides
    HomeViewContract provideHomeViewContract() {
        return mHomeViewContract;
    }
}

File: app/src/main/kotlin/me/imzack/app/end/injector/module/HomePresenterModule.kt
package me.imzack.app.end.injector.module

import dagger.Module
import dagger.Provides
import me.imzack.app.end.view.contract.HomeViewContract

@Module
class HomePresenterModule(private val mHomeViewContract: HomeViewContract) {

    @Provides
    fun provideHomeViewContract() = mHomeViewContract
}
